Code_MagneticFieldOfWire.py

```diff
@@ -0,0 +1,103 @@
+#Python code for Magnetic field of wire
+import math
+
+def magnetic_field(I, d):
+    """
+    #Calculate the magnetic field around a long straight current-carrying wire.
+
+    #Parameters:
+    I (float): Current in amperes
+    d (float): Distance from the wire in meters
+
+    Returns:
+    float: Magnetic field in teslas
+    """
+    mu_0 = 4 * math.pi * 10**-7  
+    # Permeability of free space (T m/A)
+    B = (mu_0 * I) / (2 * math.pi * d)  
+    # Magnetic field formula
+    return B
+
+# Example used:
+current = 10.0  
+# Current in amperes
+distances = (0.01, 0.1, 0.2, 0.3, 0.4, 0.5, 100)   
+# Distances in meters
+
+# Calculate and print the magnetic field for each distance
+for distance in distances:
+    B_field = magnetic_field(current, distance)
+    print(f"The magnetic field at {distance} m from a wire carrying {current} A is {B_field:.2e} T.")
+
+
+#Python code for Magnetic field of wire Using Lambda function
+import numpy as np
+import matplotlib.pyplot as plt
+
+# Constants
+mu_0 = 4 * np.pi * 10**-7  # Permeability of free space (T m/A)
+
+# Parameters
+current = 10  # Current in amperes
```

Comment:
reuse of variable without prior use

Reply:
I have changed into one time use of variable.